src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -235,6 +240,8 @@
     private boolean lastImportStatic;
     /** Whether there was any imports. */
     private boolean beforeFirstImport;
+    /** Whether static imports are independent. **/
+    private boolean staticImportsIndependent;
```

Comment:
This is only defined once and is set based on the property `option` which also doesn't change.
Is there a reason this should stay as a field? If yes, then what do you mean by independent?

Reply:
```
if (staticImportsIndependent)
```
It more readable then
```
if (option == ImportOrderOption.TOP || option == ImportOrderOption.BOTTOM)
```
That is the only reason.

Reply:
It may be simplier to read, but it still doesn't tell me the meaning behind the variable and why these 2 options are targeted. I don't know this code that well so I can't say what a better name would be.

> What about staticImportsApart?

No matter what name is used, the javadoc has to be updated to explain it a bit more.

Reply:
done